LiveMapDashboard.Web/wwwroot/js/map/park-place-markers-maplibre.js

```diff
@@ -66,18 +64,56 @@ function getSelectedCategoryIconName() {
 
 function placeMarkerOnMap(shouldCenter) {
     // If a marker already exists, remove it before adding a new one
+    //REMOVE THE IF-STATEMENT BELOW TO ESSENTIALLY ACTIVATE CLUSTERING AND MULTIPLE POIS(Although the last added POI will only be saved :P)
     if (markers.length > 0) {
         markers[0].remove(); // Remove the existing marker
         markers.length = 0; // Clear the markers array
+        pois.length = 0; // Clears the POI array
     }
 
     // Get the selected category's icon name
-    const iconName = getSelectedCategoryIconName();
+    let iconName = getSelectedCategoryIconName();
+
+    if (!iconName || iconName == "-") {
+        iconName = null;
+    }
 
     // Create a custom marker element
     const markerElement = document.createElement('div');
     markerElement.className = 'custom-marker';
 
+    const newPoi = {
+        guid: crypto.randomUUID(),
+        coordinate: { longitude: clickedLngLat.lng, latitude: clickedLngLat.lat },
+        category: {
+            iconName,
+            category: iconName ? document.querySelector('[id$="Category"]').value : 'default',
+            categoryName: iconName
+                ? document.querySelector('[id$="Category"] option:checked').text
+                : 'Uncategorized'
+        }
+    };
+    pois.push(newPoi);
+
+    // Re‑set the GeoJSON
+    const updatedGeoJson = {
+        type: 'FeatureCollection',
+        features: pois.map(poi => ({
+            type: 'Feature',
+            geometry: {
+                type: 'Point',
+                coordinates: [poi.coordinate.longitude, poi.coordinate.latitude]
+            },
+            properties: {
+                guid: poi.guid,
+                iconName: poi.category.iconName,
+                category: poi.category.category,
+                categoryName: poi.category.categoryName
+            }
+        }))
+    };
+    map.getSource('pois').setData(updatedGeoJson);
```

Comment:
```suggestion
   if (map.getSource('pois')) {
        map.getSource('pois').setData(updatedGeoJson);
    }
```

Deze functie wordt niet altijd geroepen na de load van map. Als ik me niet vergis wordt dit ook gecalled als categorie veranderd. Gezien de source pas op map load gezet wordt gaf dit bij mij een exception waardoor alles faalde